t/lib/redis.php

```diff
@@ -103,62 +103,63 @@ private function read_data() {
 
-        if ($prefix == 36) { // '$' 
-            $size = intval(substr($data, 1));
-            if ($size < 0) {
-                return null;
-            }
+        switch(ord($data[0])) {
+            case 36: // '$'
+                $size = intval(substr($data, 1));
+                if ($size < 0) {
+                    return null;
+                }
+                
+                $token = strtok($data, "\r\n");
+                $token = strtok("\r\n");
+                return $token;
+
+            case 43: // '+'
+                return trim(substr($data, 1));
             
-            $token = strtok($data, "\r\n");
-            $token = strtok("\r\n");
-            return $token;
-
-        }else if ($prefix == 43) { // '+'
-            return trim(substr($data, 1));
-        }else if ($prefix == 42) { // '*'
-            $size = intval(substr($data, 1));
-            if ($size < 0) {
-                return null;
-            }
-
-            $output = array();
-            $data = substr($data, strpos($data, "\r\n")+2);
-            for($i = 0; $i < $size; $i++) {
-                $res = $this->data_parse($data);
-                #var_dump($data);
-                if (!empty($res)) {
-                    $output[$i] = $res;
-                }else {
-                    $output[$i] = null;
+            case 42: // '*'
+                $size = intval(substr($data, 1));
+                if ($size < 0) {
+                    return null;
                 }
-                $data = substr($data, strpos($data, "\r\n")+2);
-                $data = substr($data, strpos($data, "\r\n")+2);
-            }
-            return $output;
 
-        }else if ($prefix == 58) { // ':'
-            return intval(substr($data, 1));
-        }else if ($prefix == 45) { // '-'
-            return false;
-        }else {
-            return null;
+                $output = array();
+                $data = substr($data, strpos($data, "\r\n")+2);
+                for($i = 0; $i < $size; $i++) {
+                    $res = $this->data_parse($data);
+                    #var_dump($data);
+                    if (!empty($res)) {
+                        $output[$i] = $res;
+                    }else {
+                        $output[$i] = null;
+                    }
+                    $data = substr($data, strpos($data, "\r\n")+2);
+                    $data = substr($data, strpos($data, "\r\n")+2);
+                }
+                return $output;
+            
+            case 58: // ':'
+                return intval(substr($data, 1));
+            
+            case 45: // '-'
+                return false;
+            
+            default:
+                return null;
         }
     }
 
     public function __call($name, $params) {
-        if (!in_array($name, $this->commands)) {
+        if (!in_array($name, self::$commands)) {
             return false;
         }
 
         array_unshift($params, $name);
         #var_dump($params);
```

Comment:
couldnt this be outputed if a debug value of some kind is set?

Reply:
Thanks

Reply:
Please could you create a new issue about that,
Thank you.